uuid.go

```diff
@@ -156,6 +158,47 @@ func (u UUID) String() string {
 	return string(buf)
 }
 
+// Format implements fmt.Formatter for UUID values.  It accepts 'h' or 'H' as a format to output only
+// the hex digits of the UUID.  Any other format specifier returns the canonical RFC-4122 representation.
+func (u UUID) Format(f fmt.State, c rune) {
+	//
+	// TODO: dylan.bourque - 2019-02-15
+	// . should we implement %v? if so, what is "a Go-syntax representation" of a UUID when using %#v?
+	//
+	switch c {
+	case 'h', 'H':
+		s := hex.EncodeToString(u.Bytes())
+		if c == 'H' {
+			s = strings.Map(toCapitalHexDigits, s)
+		}
+		_, _ = io.WriteString(f, s)
+	case 'q':
```

Comment:
I don't think this is necessary. I think we're getting this behavior out of the box: https://play.golang.org/p/GcHtLq7aZxl

The `fmt` package documentation also talks about this: 

> If the format (which is implicitly %v for Println etc.) is valid for a string (%s %q %v %x %X), the following two rules apply:
> 
> 4. If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).
> 
> 5. If an operand implements method String() string, that method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

Reply:
I tested this case and you must account for the %q verb in your `.Format()` implementation.  If you don't handle it directly, the result is whatever you do in the default case.  See here -> https://play.golang.org/p/FNja_rxReEx

Reply:
Ah. I was not clear enough, sorry. What I meant to say is that if we're not implementing `fmt.Formatter`, we're getting this behavior for `%q` anyway, due to `UUID` implementing `fmt.Stringer` and the rule I quoted above.

But if we _do_ implement `fmt.Formatter`, we should honor `%q` explicitly indeed.